### PR TITLE
CHE-1004: Improve experience on Debugger configuration selection

### PR DESCRIPTION
**Pull request by azatsarynnyy**
- added facade for debug configuration related operations (DebugConfigurationsManager) in Debugger API;
- drop-down list for selecting debug configuration removed from toolbar;
- all debug configurations are available for selecting as actions in Run -> Debug Configurations menu.

@vparfonov

### CODE REVIEW

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    private List<DebugConfigurationDto> retrieveConfigurations() {
+        List<DebugConfigurationDto> configurationsList;
+
+        if (localStorageOptional.isPresent()) {
+            final LocalStorage localStorage = localStorageOptional.get();
+            final Optional<String> data = Optional.fromNullable(localStorage.getItem(LOCAL_STORAGE_DEBUG_CONF_KEY));
+            if (data.isPresent() && !data.get().isEmpty()) {
+                configurationsList = dtoFactory.createListDtoFromJson(data.get(), DebugConfigurationDto.class);
+            } else {
+                configurationsList = new ArrayList<>(0);
```
Comment (skabashnyuk) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java]: emptyList()

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            }
+        } else {
+            configurationsList = new ArrayList<>(0);
```
Comment (skabashnyuk) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java]: emptyList()

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+ */
+public class DebugConfigurationsManagerImpl implements DebugConfigurationsManager {
+
+    private static final String LOCAL_STORAGE_DEBUG_CONF_KEY = "che-debug-configurations";
+
+    private final DtoFactory                        dtoFactory;
+    private final DebugConfigurationTypeRegistry    configurationTypeRegistry;
+    private final Optional<LocalStorage>            localStorageOptional;
+    private final Set<ConfigurationChangedListener> configurationChangedListeners;
+    private final List<DebugConfiguration>          configurations;
```
Comment (skabashnyuk) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java:47]: Will it be more efficient to store configurations in Map?

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java (hunk 1 above)
Comment (azatsarynnyy) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java]: fixed

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java (hunk 2 above)
Comment (azatsarynnyy) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java]: thx
fixed

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java (hunk 3 above)
Comment (azatsarynnyy) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/configuration/DebugConfigurationsManagerImpl.java:47]: I think that no need to use Map here

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/debug/DebugConfigurationsManager.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    void setCurrentDebugConfiguration(@Nullable DebugConfiguration debugConfiguration);
+
+    /** Returns all debug configurations. */
+    List<DebugConfiguration> getConfigurations();
+
+    /**
+     * Creates new configuration with the given parameters.
+     *
+     * @return created {@link DebugConfiguration}
+     * @throws IllegalArgumentException
```
Comment (svor) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/debug/DebugConfigurationsManager.java]: The method isn't throwing  any exception

Comment (azatsarynnyy) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/debug/DebugConfigurationsManager.java]: thx, fixed

**Pull request merged**